### PR TITLE
Add `unist-util-ancestor` to list of utilities

### PR DESCRIPTION
**Pull request by gorango**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

Adding a small utility to `readme` that helps with finding common ancestors of one or more unist nodes. It has come in handy a few times while working with ast trees so I thought I'd package it up. Here's hoping it might be helpful to someone else too!

P.S. I am open to renaming the package if desired.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -431,6 +431,8 @@ unist:
 
 ### List of utilities
 
+*   [`unist-util-ancestor`](https://github.com/gorange/unist-util-ancestor)
```
Comment (wooorm) [inline readme.md:434]:
```suggestion
*   [`unist-util-ancestor`](https://github.com/gorango/unist-util-ancestor)
```

**Comment by wooorm** [comment]
Nice!

For performance reasons you might want to not use `unist-util-find` (in this way). And perhaps also `getParents` like this.
Right now, for every node, the whole tree is walked many many times.

If possible, it would be muuuch faster to walk the tree once/twice.
Here’s some pseudocode:

```js
const stacks = new Map()

visitParents(tree, (node, parents) => {
  if (nodesToFind.includes(node)) {
    stacks.set(node, parents)
  }
})

nodesToFind.forEach(node => {
  if (!stacks.has(node)) {
    throw new Error('unist-util-ancestor requires all nodes be contained in the tree')
  }
})

// Somethign like this?
let index = -1
let ancestor = tree

// This is broken, maybe you get the gist?
while (++index) {
  const nextAncestor = stacks.get(nodesToFind[0])[index]

  const shared = nodesToFind.every(node => stacks.get(node)[index] === nextAncestor)

  // To do: exit if there are no nodes left on stacks.

  if (shared) {
    ancestor = nextAncestor
  } else {
    break
  }
}
```

**Comment by gorango** [comment]
Thank you so much for taking the time to review and provide this great feedback!

I have implemented your suggestions and removed the find util completely. Way cleaner :)

**Comment by wooorm** [comment]
That looks great! Nice find as well on the things that I didn't get to. Glad it worked!

**Pull request merged**